src/core/MOM_open_boundary.F90

```diff
@@ -5080,13 +5081,21 @@ subroutine update_segment_tracer_reservoirs(G, GV, uhr, vhr, h, OBC, dt, Reg)
                           ! For salinity the units would be [ppt S-1 ~> 1]
   integer :: i, j, k, m, n, ntr, nz
   integer :: ishift, idir, jshift, jdir
+  real :: b_in, b_out     ! The 0 and 1 switch for tracer reservoirs
+                          ! 1 if the length scale of reservoir is zero [nodim]
+  real :: a_in, a_out     ! The 0 and 1(-1) switch for reservoir source weights
+                          ! e.g. a_in is -1 only if b_in ==1 and uhr or vhr is inward
+                          ! e.g. a_out is 1 only if b_out==1 and uhr or vhr is outward
+                          ! It's clear that a_in and a_out cannot be both non-zero [nodim]
 
   nz = GV%ke
   ntr = Reg%ntr
 
   if (associated(OBC)) then ; if (OBC%OBC_pe) then ; do n=1,OBC%number_of_segments
     segment=>OBC%segment(n)
     if (.not. associated(segment%tr_Reg)) cycle
+    b_in  = 0; if (segment%Tr_InvLscale_in  == 0) b_in  = 1
```

Comment:
In this case, the compilers will do the integer-to-real conversions to do the right thing, but when setting a real variable to 0 or 1, or comparing a real value to 0, the preferred syntax would be `b_in = 0.0 ; `, rather than `b_in = 0; `, and similarly for the other 5 instances in these two lines.